### PR TITLE
Added pip update description for Ubuntu/Debian

### PR DESCRIPTION
**Pull request by meriac**

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -114,7 +114,14 @@ First install yotta's dependencies using your system's package manager, for
...
 
 ```sh
+# install development tool dependencies
 sudo apt-get install python-pip cmake build-essential ninja-build python-dev
+
+# update pip to latest release
+sudo apt-get remove python-pip
+curl -o get-pip.py https://bootstrap.pypa.io/get-pip.py
+sudo python get-pip.py
+hash -r
```
Comment (autopulated) [inline docs/index.md]: what's `hash -r` for?

Comment (meriac) [inline docs/index.md]: It's for re-hashing the search path - Ubuntu is too stupid to figure out that pip changed its location otherwise.

Comment (danros) [inline docs/index.md]: Isn't executing some arbitrary downloaded python script as root somewhat of an anti-pattern?

Should we not be adding some other repository to provide the latest pip, or using easy_install ?

Comment (meriac) [inline docs/index.md]: Yeah, what can you do - these long haired "believe-in-the-good-of-mankind" open source developers :-).

It's not superbad, as curl verifies at least that the SSL is certified by someone who is allowed to run abitrary code on your machine (AKA your friendly HCHQ / NSA case person et al).

It's not really fixable from the outside (http://pip.readthedocs.org/en/latest/installing.html) - as you usually want the latest version - unless we want to maintain a mirror.

Creating a yotta package for pip would be a catch-22.

Same with using a SHA256 checksum etc - once the file changes we're toast.

The RPM and DEB pakages with ignatures are hopelssly outdated (ubunto, Fedora). Unless you have a better idea I suggest to stick with my suggested patch.

Comment (danros) [inline docs/index.md]: Must admit that I haven't looked into why yotta requires the very latest version of pip. However it's just worth mentioning that i've never had to do something like this in many years of using python.

Comment (autopulated) [inline docs/index.md]: I haven't had any problems, so I don't know. I don't think it does require the latest version, just not an ancient one.

**Comment by autopulated** [comment]
I think it should no longer be necessary to update pip (it was an issue with one of the dependencies). Can you confirm if this is the case? If it is, I'd rather keep the instructions simple, and not tell people to update pip.

**Comment by autopulated** [comment]
Closing because I believe this is no longer necessary.

**Pull request closed without merging**